library/src/main/java/com/datatorrent/lib/io/block/BlockMetadata.java

```diff
@@ -216,20 +217,45 @@ public FileBlockMetadata(String filePath, long blockId, long offset, long length
       this.filePath = filePath;
     }
 
+    public FileBlockMetadata(String filePath, long blockId, long offset, long length, boolean isLastBlock,
+        long previousBlockId, long fileLength)
+    {
+      super(blockId, offset, length, isLastBlock, previousBlockId);
+      this.filePath = filePath;
+      this.fileLength = fileLength;
+    }
+
     public FileBlockMetadata(String filePath)
     {
       this.filePath = filePath;
     }
 
+    public FileBlockMetadata(String filePath, long fileLength)
+    {
+      this.filePath = filePath;
+      this.fileLength = fileLength;
+    }
+
     public String getFilePath()
     {
       return filePath;
     }
 
+    public long getFileLength()
+    {
+      return fileLength;
+    }
+
     public FileBlockMetadata newInstance(@NotNull String filePath)
     {
       Preconditions.checkNotNull(filePath);
       return new FileBlockMetadata(filePath);
     }
+
+    public FileBlockMetadata newInstance(@NotNull String filePath, long fileLength)
+    {
```

Comment:
We had newInstance() method with filePath. Since we have added the new parameter, I thought of creating this method as well to support both filePath and fileLength